elements/lisk-chain/test/unit/block.spec.ts

```diff
@@ -228,4 +228,186 @@ describe('block', () => {
 			});
 		});
 	});
+
+	describe('validateGenesis', () => {
+		let block: Block;
+		let tx: Transaction;
+		let assetList: BlockAsset[];
+		let blockAssets: BlockAssets;
+
+		beforeEach(() => {
+			assetList = [
+				{
+					moduleID: 3,
+					data: getRandomBytes(64),
+				},
+				{
+					moduleID: 6,
+					data: getRandomBytes(64),
+				},
+			];
+			blockAssets = new BlockAssets(assetList);
+			tx = getTransaction();
+		});
+
+		describe('when all values are valid', () => {
+			it('should not throw error', async () => {
+				// Arrange
+				block = await createValidDefaultBlock({ payload: [], assets: blockAssets });
+				// Act & assert
+				expect(() => block.validateGenesis()).not.toThrow();
+			});
+		});
+
+		describe('when payload is not empty', () => {
+			it('should throw error', async () => {
+				// Arrange
+				const txs = new Array(20).fill(0).map(() => tx);
+				block = await createValidDefaultBlock({ payload: txs, assets: blockAssets });
+				// Act & assert
+				expect(() => block.validateGenesis()).toThrow('Payload length must be zero');
+			});
+		});
+
+		describe('when an asset data has size more than the limit', () => {
```

Comment:
this should only applied to the normal block?

Reply:
I think here it is just checking that there is no limitation to the `data` property i.e. it does not throw error